rllib/utils/actor_manager.py

```diff
@@ -790,17 +806,27 @@ def probe_unhealthy_actors(
             mark_healthy: Whether to mark actors healthy if they respond to the ping.
 
         Returns:
-            A list of actor ids that are restored.
+            A list of actor IDs that were restored by the `ping` AND those actors that
+            were previously restored via other remote requests. The cached set of
+            such previously restored actors will be erased in this call.
         """
+        # Collect recently restored actors (from `self.__fetch_result` calls other than
+        # the one triggered here via the `ping`).
+        restored_actors = list(self.__restored_actors)
```

Comment:
This was actually a bug:
The `Algorithm.on_workers_recreated` callback would NOT fire in case an actor got restarted b/c of another remote request made to it (other than the `ping()` in this method here, which called once per training iteration).